code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm

```diff
@@ -1529,3 +1529,147 @@
+					Androrovir: Will transform someone's sex to male. <br>
+					Gynorovir: Will transform someone's sex to female. <br>
+					Androgynorovir: Will transform someone's sex to plural. <br>
+					<br>
+					<B>Special Chemicals</B><BR>
+					Stoxin: Will make someone drowsy. <br>
+					Rainbow Toxin: Will make someone see rainbows. <br>
+					Paralysis Toxin: Will make someone paralyzed. <br>
+					Numbing Enzyme: Will make someone unable to feel pain. <br>
+					Pain Enzyme: Will make someone feel amplified pain. <br>
+					<br>
+					<B>Side Notes</B><BR>
+					You can select a value of 0 to inject nothing! <br>
+					Overdose threshold for most chemicals is 30 units. <br>
+					Exceptions to OD is: (Numbing Enzyme:20)
+					<br>
+					"}
+		usr << browse(output,"window=chemicalrefresher")
+		return
+	else
+		var/list/targets = list() //IF IT IS NOT BROKEN. DO NOT FIX IT. AND KEEP COPYPASTING IT  (Pointing Rick Dalton: "That's my code!" ~CL)
+
+		for(var/mob/living/carbon/L in living_mobs(1, TRUE)) //Noncarbons don't even process reagents so don't bother listing others.
+			if(!istype(L, /mob/living/carbon))
+				continue
+			if(L == src) //no getting high off your own supply, get a nif or something, nerd.
+				continue
+			if(!L.resizable && (trait_injection_selected == "macrocillin" || trait_injection_selected == "microcillin" || trait_injection_selected == "normalcillin")) // If you're using a size reagent, ignore those with pref conflicts.
+				continue
+			if(!L.allow_spontaneous_tf && (trait_injection_selected == "androrovir" || trait_injection_selected == "gynorovir" || trait_injection_selected == "androgynorovir")) // If you're using a TF reagent, ignore those with pref conflicts.
+				continue
+			targets += L
+
+		if(!(targets.len))
+			to_chat(src, "<span class='notice'>No eligible targets found.</span>")
+			return
+
+		var/mob/living/target = tgui_input_list(src, "Please select a target.", "Victim", targets)
+
+		if(!target)
+			return
+
+		if(!istype(target, /mob/living/carbon)) //Safety.
+			to_chat(src, "<span class='warning'>That won't work on that kind of creature! (Only works on crew/monkeys)</span>")
+			return
+
+		if(target.isSynthetic())
+			to_chat(src, "<span class='notice'>There's no getting past that outer shell.</span>")
+			return
+
+		if(!trait_injection_selected)
+			to_chat(src, "<span class='notice'>You need to select a reagent.</span>")
+			return
+
+		if(!trait_injection_verb)
+			to_chat(src, "<span class='notice'>Somehow, you forgot your means of injecting. (Select a verb!)</span>")
+			return
+
+		if(do_after(src, 50, target)) //A decent enough timer.
+			add_attack_logs(src,target,"Injection trait ([trait_injection_selected], [trait_injection_amount])")
+			if(target.reagents && (trait_injection_amount > 0))
+				target.reagents.add_reagent(trait_injection_selected, trait_injection_amount)
+			var/ourmsg = "<span class='warning'>[usr] [trait_injection_verb] [target] "
+			switch(zone_sel.selecting)
```

Comment:
arent those defines in the switch just strings? Cant it be just all  if(zone_sel.selecting) ourmsg += (zone_sel.selecting)?

Reply:
Ah, nearly forgot, then it's fine.